[csv/en] New language CSV

turbotimon opened the pull request:
- [x] I solemnly swear that this is all original content of which I am the original author
- [x] Pull request title is prepended with `[language/lang-code]` (example `[python/fr-fr]` or `[java/en]`)
- [x] Pull request touches only one file (or a set of logically related files with similar changes made)
- [x] Content changes are aimed at *intermediate to experienced programmers* (this is a poor format for explaining fundamental programming concepts)
- [x] If you've changed any part of the YAML Frontmatter, make sure it is formatted according to [CONTRIBUTING.md](https://github.com/adambard/learnxinyminutes-docs/blob/master/CONTRIBUTING.markdown)
  - [x] Yes, I have double-checked quotes and field names!

Closes #5202

nbehrnd commented:
@turbotimon Try to stay below the 80 character limit `CONTRIBUTING.markdown` recommends.  One reason for this is to ease reading git's diff views.  Either set up your text editor to display an indicator, request automatic hard line breaks, or fold the file.

For the former and presuming Windows as operating system available to you, [Notepad++](https://notepad-plus-plus.org/) allows you to set one or multiple vertical marks.  Go Settings -> Preferences -> Margins/Border/Edge -> Vertical edge settings (centre of the widget).  In case you have access to [vim](https://www.vim.org/) (an editor more prominent in Unix/Linux, but cross-platform (e.g., [gvim](https://portableapps.com/apps/development/gvim_portable)) add `set colorcolumn=80` to your `.vimrc` configuration file to the same effect.

A Linux shell (e.g., the Git Bash by [tortoise git](https://tortoisegit.org/) in Windows) can equally help you to provide the edit _post factum_:

- run `fold -s input.file > temp` to break lines longer than 80 characters.  Instead of a cutting words which pass across this threshold, the optional `-s` identifies the latest white space prior to this limit to do so.
- next, run `sed 's/ \+$//' temp > out` to write an `out` file without trailing white spaces the first step usually creates.  If it passes the visual check, it can overwrite your submission.

A single blank line (after the first bullet list, line 16) is enough, too.

nbehrnd commented:
@turbotimon  It is true, the 80 characters limit is somewhat arbitrary, but to help the layout once the markdown files are rendered, too.  To prevent the need to scroll horizontally (e.g., current day's rendering about [ReST](https://learnxinyminutes.com/docs/rst/)).  For milder cases (like here), by the browser's set up to use locally installed fonts / to optionally overrule the suggestions sent by a web site, a single character needn't be the camel's straw.  It depends.  The hint to markdownlint is an indicator that there tools around to help.

Speaking of such, lines 29, 47, 62, 83 _appear_ like sub sections instead of being such (i.e., first defined by function, and their appearance a consequence).  A leading `##` as header of level 2 instead of enclosing them by `**` (similar to the file about markdown itself, [source](https://github.com/adambard/learnxinyminutes-docs/blob/master/markdown.html.markdown), [rendering](https://learnxinyminutes.com/docs/markdown/)) might match your intent.\*

Regarding the contents: to me, the submission is a useful read more likely to fit into "Data Structures", than "Languages".  I welcome it equally notes briefly tab separated files used for instance in the "chemistry spreadsheets" of e.g., [Datawarrior](https://openmolecules.org/index.html).

\* Though there are a few different approaches to GitHub's Markdown, I keep a copy of [Kofler's reference](https://kofler.info/ebooks/markdown_pandoc/) about Pandoc's flavor.  But there are so many markdown implementations around.

turbotimon commented:
Hi @nbehrnd 

Sorry for the silence.

> Speaking of such, lines 29, 47, 62, 83 appear like sub sections 
Agree, I changed that

> more likely to fit into "Data Structures"
Agree, that language is maybe not the best fit, not sure if Data Structure is, but i changed it. Would fit more into a seperate sektion "File format" or "Data exchange format" which would also fit more for json, yaml and so on.

> to me, the submission is a useful read
Thanks. I really like this site and I use it whenever I can. My main motivation for adding this is, that I also teach and often link to this page for a short and concise explanation of data exchange formats like json, yaml and csv. But I always missed the last one:)

nbehrnd commented:
@turbotimon To me, contents and format look good.

@verhovsky, @vendethiel Frequently, I saw you to triage and merge PRs to the project.  May you have a look on this one, please?  Can you comment on @turbotimon's idea to file this contribution (altogether with the markdown files about `json` and `yaml`) into a new section "file format" or "data exchange format"?

turbotimon commented:
any progress on this? I would like to see this merged. Any other thing (e.g. new sections) should be apart from that

vendethiel commented:
Sorry, that completely slipped my mind. It should probably be in its own category indeed.

turbotimon commented:
i suggest we merge this first and then think about appropriate categories in a separate issue.

Because "languages" is very broad. I suggest e.g. splitting in categories like "programming languages" (e.g. python), "markup languages" (e.g. asciidoc)  and "data interchange formats" (e.g. json, csv). but there are some overlaps and should be carefully thought through in a separate issue to not mix up things

nbehrnd commented:
@vendethiel a gentle reminder

vendethiel commented:
@verhovsky do you know if the backend lists categories or if it's freeform?

verhovsky commented:
There shouldn't be more categories. Calling data markup languages "languages" is fine in my opinion.

And this CSV explanation isn't following the Learn X in Y Minutes format, it should be one or two files showing the quoting rules, with less natural language explanation in between.

verhovsky commented:
It's not freeform in learnxinyminutes-docs because the order in [build.py](https://github.com/adambard/learnxinyminutes-site/blob/6c1c8a784450c4c61c28d2eeaa589ddee0b6019b/build.py#L300-L303) affects the order the categories [appear on the website](https://github.com/adambard/learnxinyminutes-site/blob/6c1c8a784450c4c61c28d2eeaa589ddee0b6019b/templates/index.html#L18).

turbotimon commented:
> And this CSV explanation isn't following the Learn X in Y Minutes format, it should be one or two files showing the quoting rules, with less natural language explanation in between.

It's hard for CSV because there are no comments allowed. I did some examples with the explanation a string in the examples which got removed now...
